From a80c1b92901a9f65751bbe9c1f122f8af8a35f3c Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Wed, 30 Apr 2025 12:57:43 +0800
Subject: [PATCH 176/254] mtk: hostapd: avoid clearing vht 160 support during
 downgrading bw

Avoid clearing BW160 support in VHT capabilities when downgrading bandwidth via channel switch.
Otherwise, the STA will store the incorrect VHT capabilities of its connected AP, which leads
to a VHT capabilities unsync issue.
In this case, the STA will not be able to follow the AP to switch from BW80 back to BW160,
since the VHT capabilities of the AP stored by the STA indicate that BW160 is not supported.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 src/ap/drv_callbacks.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index 07c4c98e0..b3f8ada16 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -1392,7 +1392,6 @@ void hostapd_event_ch_switch(struct hostapd_data *hapd, int freq, int ht,
 	hapd->iconf->punct_bitmap = punct_bitmap;
 #endif /* CONFIG_IEEE80211BE */
 	if (hapd->iconf->ieee80211ac) {
-		hapd->iconf->vht_capab &= ~VHT_CAP_SUPP_CHAN_WIDTH_MASK;
 		if (chwidth == CONF_OPER_CHWIDTH_160MHZ)
 			hapd->iconf->vht_capab |=
 				VHT_CAP_SUPP_CHAN_WIDTH_160MHZ;
-- 
2.45.2

